test/artifacts.jl

```diff
@@ -464,17 +460,13 @@ end
         generate_flooblegrank_artifacts(ap_path)
 
         Pkg.activate(ap_path)
-        add_this_pkg()
-        @test !isdir(artifact_path(engaged_hash))
-        @test !isdir(artifact_path(disengaged_hash))
```

Comment:
I don't understand what operation is installing the flooblecrank artifacts here.

Reply:
After we add a package we also make sure that everything is instantiated. That has been there for a long time. But there was a bug where artifacts for the current active package did not get installed, but that is fixed now. So when we do `add_this_pkg`, it instantiates and that installs the flubidooki artifact of the current package.

Reply:
The bug fix is basically https://github.com/JuliaLang/Pkg.jl/pull/2268/files#diff-6dfb4e40705b22a427451362ea83b4e3a763028e3dcaa3b9b1846aa1ec84370fR596-R597.

Reply:
That makes sense to me, but I don't see where the `add_this_pkg()` is; you've removed it.

Reply:
It is on (the new) line 469:

https://github.com/JuliaLang/Pkg.jl/pull/2268/files#diff-bf3b8e7b503ee397f9d6ba823767c04bf212d23b4669c50d9753e6897d9c1ecbR469

Reply:
But that is below these lines?  So wouldn't these `@test` lines still pass?  The intent here was to verify that these two artifacts are not present, so we can assert that they were added within those `withenv()` blocks, and not accidentally added by something else earlier in the test recipe.

Reply:
I see, yes, that makes sense. I should be able to add these back indeed.

Reply:
Great!  Then we're totally on the same page.  :)